### PR TITLE
feat(HackTheBox): add presence

### PR DESCRIPTION
**Pull request by Atsukoro1**
## Description 
Adds presence for HackTheBox (app.hackthebox.com)

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

**Review by EncryptedDev — COMMENTED** [review]
Remove all changes unrelated to this specific PR

File: websites/H/HackTheBox/metadata.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+	"$schema": "https://schemas.premid.app/metadata/1.7",
+	"author": {
+		"id": "937757295453044806",
+		"name": "Atsukoro1#6323"
```
Comment (theusaf) [inline websites/H/HackTheBox/metadata.json:5]: Remove discriminator from old `pmd`
```suggestion
		"name": "Atsukoro1"
```

File: websites/H/HackTheBox/metadata.json [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	},
+	"service": "HackTheBox",
+	"description": {
+		"en": "Hack The Box is a massive, online cybersecurity training platform, allowing individuals, companies, universities and all kinds of organizations around the world to level up their hacking skills."
+	},
+	"url": "app.hackthebox.com",
+	"version": "1.0.0",
+	"logo": "https://i.imgur.com/eNi7hX2.png",
```
Comment (theusaf) [inline websites/H/HackTheBox/metadata.json:13]: Logos must be [512x512px at least](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#logo). Find or create a larger version.

File: websites/H/HackTheBox/presence.ts [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+		if (item) {
+			presenceData.details = item.details;
+			if (item.stateMethod)
+				presenceData.state = methods[item.stateMethod as keyof IMethods]();
+			else presenceData.state = item.state;
+		}
+	}
+
+	presenceData.largeImageKey = "icon";
```
Comment (theusaf) [inline websites/H/HackTheBox/presence.ts:205]: Imgur links are preferred over Discord asset names.

File: websites/H/HackTheBox/presence.ts [hunk 4]
```diff
@@ -0,0 +1,218 @@
...
+	presenceData.buttons = [
+		{
+			label: "View Page",
+			url: document.location.href,
+		},
```
Comment (theusaf) [inline websites/H/HackTheBox/presence.ts:210]: This has the possibility of adding a button that leads to the home page, which is [not allowed](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md#presencets). Adding a button to every single page is not good practice for this repo.

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/H/HackTheBox/presence.ts [hunk 5]
```diff
@@ -0,0 +1,215 @@
+const presences = [
```
Comment (EncryptedDev) [inline websites/H/HackTheBox/presence.ts:1]: This should be a Record<string, PresenceData> instead
https://github.com/PreMiD/Presences/blob/236523980322a4053a45c691a1739eb50e2aea48/websites/G/Geotastic/presence.ts#L12-L42

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

File: websites/H/HackTheBox/presence.ts (hunk 5 above)
Comment (Atsukoro1) [reply websites/H/HackTheBox/presence.ts:1]: Yeah, that's great, but it needs to be somehow extended by the stateMethod I have there.

**Comment by EncryptedDev** [comment]
Updated title to follow commitlint expectation which you agreed to in the prerequisites.

File: websites/H/HackTheBox/presence.ts [hunk 6]
```diff
@@ -0,0 +1,188 @@
+const presences: Record<string, PresenceData> = {
+		"/challenges/retired": {
+			details: "Challenges",
+			state: "Browsing retired challenges...",
```
Comment (EncryptedDev) [inline websites/H/HackTheBox/presence.ts:4]: Remove `...` in all strings

File: websites/H/HackTheBox/presence.ts [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+	}`;
+}
+
+function getPersonalProfileDetails() {
+	return document.querySelectorAll(".htb-subtitle.greenOnHover")[0].textContent;
+}
+
+function getMachineDetails() {
+	const diff = document.querySelectorAll(
+			".d-inline-block.pl-2.pr-2.borderRadius13.label-reset.no-space.color-main.htb-label.undefined.letterSpacingundefined.borderRadius4"
```
Comment (EncryptedDev) [inline websites/H/HackTheBox/presence.ts:100]: Shorten this

File: websites/H/HackTheBox/presence.ts [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+		)[0].textContent,
+		name = document
+			.querySelectorAll(
+				".d-inline-block.pl-2.pr-2.borderRadius13.label-reset.no-space.color-white.fontSemiBold.letterSpacing0.borderRadius4"
```
Comment (EncryptedDev) [inline websites/H/HackTheBox/presence.ts:104]: Ditto

File: websites/H/HackTheBox/presence.ts [hunk 9]
```diff
@@ -0,0 +1,188 @@
...
+			? "offline"
+			: "online";
+
+	return `${name} (${diff}) - ${status}`;
+}
+
+function getChallengeDetails() {
+	const name = document
+		.querySelectorAll(
+			".d-inline-block.pl-2.pr-2.borderRadius13.label-reset.no-space.color-white.fontSemiBold.letterSpacing0.borderRadius4"
```
Comment (EncryptedDev) [inline websites/H/HackTheBox/presence.ts:119]: Ditto

File: websites/H/HackTheBox/presence.ts [hunk 10]
```diff
@@ -0,0 +1,188 @@
...
+		default:
+			return "";
+	}
+}
+
+presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {};
+
+	for (const [path, data] of Object.entries(presences)) {
+		const regex = new RegExp(path.replace(/{}/g, ".*"), "g");
```
Comment (EncryptedDev) [inline websites/H/HackTheBox/presence.ts:157]: Resolve CI here

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Remove changes to lock file too

File: websites/H/HackTheBox/presence.ts [hunk 11]
```diff
@@ -0,0 +1,185 @@
...
+presence.on("UpdateData", async () => {
+	let presenceData: PresenceData = {};
+
+	for (const [path, data] of Object.entries(presences)) {
+		presenceData = {
+			...presenceData,
+			largeImageKey: "https://i.imgur.com/aMjnyic.png",
+			buttons: [
+				{
+					label: "My Profile",
```
Comment (Bas950) [inline websites/H/HackTheBox/presence.ts:159]:
```suggestion
					label: "View Profile",
```

**Review by Bas950 — CHANGES_REQUESTED** [review]

**Review by Bas950 — APPROVED** [review]

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Remove package-lock changes

File: websites/H/HackTheBox/presence.ts [hunk 12]
```diff
@@ -0,0 +1,185 @@
...
+			.textContent
+	}`;
+}
+
+function getPersonalProfileDetails() {
+	return document.querySelectorAll(".htb-subtitle.greenOnHover")[0].textContent;
+}
+
+function getMachineDetails() {
+	const container = document.querySelectorAll(".text-left.pl-8.pt-3")[0],
```
Comment (theusaf) [inline websites/H/HackTheBox/presence.ts:103]: These `.querySelectorAll(...)[0]` can be replaced with `.querySelector(...)`

File: websites/H/HackTheBox/presence.ts [hunk 13]
```diff
@@ -0,0 +1,185 @@
...
+		case "/home":
+			return getHomePageDetails();
+		case "/profile/overview":
+			return getPersonalProfileDetails();
+		case "/machines/{}":
+			return getMachineDetails();
+		case "/challenges/{}":
+			return getChallengeDetails();
+		case "/users/{}":
+			return "";
```
Comment (theusaf) [inline websites/H/HackTheBox/presence.ts:144]:
```suggestion
```
Handled by default case

**Review by theusaf — CHANGES_REQUESTED** [review]

File: websites/H/HackTheBox/presence.ts [hunk 14]
```diff
@@ -0,0 +1,183 @@
...
+		},
+	},
+	presence = new Presence({
+		clientId: "1042105891681472595",
+	});
+
+function getHomePageDetails() {
+	return `${
+		document
+			.querySelectorAll(".text-left.font-size15.col")[0]
```
Comment (theusaf) [inline websites/H/HackTheBox/presence.ts:90]: `.querySelectorAll(...)[0]` is still being used a lot where it could be replaced by `.querySelector(...)`

**Review by theusaf — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Pull request merged**